<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>803</width>
    <height>516</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <widget class="QWidget" name="centralWidget">
   <layout class="QGridLayout" name="gridLayout">
    <item row="8" column="0" colspan="2">
     <widget class="QGroupBox" name="groupBox">
      <property name="title">
       <string>GCode Stats</string>
      </property>
      <layout class="QFormLayout" name="formLayout">
       <item row="0" column="0">
        <widget class="QLabel" name="label_3">
         <property name="text">
          <string>Number of line</string>
         </property>
        </widget>
       </item>
       <item row="0" column="1">
        <widget class="QLabel" name="NumOfLine">
         <property name="text">
          <string/>
         </property>
         <property name="alignment">
          <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
         </property>
        </widget>
       </item>
      </layout>
     </widget>
    </item>
    <item row="7" column="0" colspan="2">
     <widget class="QGroupBox" name="groupBox_2">
      <property name="title">
       <string>Part Stats</string>
      </property>
      <layout class="QFormLayout" name="formLayout_2">
       <property name="fieldGrowthPolicy">
        <enum>QFormLayout::AllNonFixedFieldsGrow</enum>
       </property>
       <item row="0" column="0">
        <widget class="QLabel" name="label_5">
         <property name="text">
          <string>Number of layers</string>
         </property>
        </widget>
       </item>
       <item row="0" column="1">
        <widget class="QLabel" name="NumOfLayer_lbl">
         <property name="text">
          <string/>
         </property>
         <property name="alignment">
          <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
         </property>
        </widget>
       </item>
      </layout>
     </widget>
    </item>
    <item row="0" column="5" rowspan="20">
     <widget class="QMdiArea" name="MDI_Area"/>
    </item>
    <item row="1" column="1">
     <widget class="QToolButton" name="SelectFileGeom_btn">
      <property name="text">
       <string>...</string>
      </property>
     </widget>
    </item>
    <item row="0" column="0" colspan="2">
     <widget class="QLabel" name="label">
      <property name="text">
       <string>Geometry file</string>
      </property>
     </widget>
    </item>
    <item row="3" column="0" colspan="3">
     <widget class="QDoubleSpinBox" name="Layer_num">
      <property name="minimum">
       <double>0.500000000000000</double>
      </property>
      <property name="maximum">
       <double>3.000000000000000</double>
      </property>
      <property name="value">
       <double>1.000000000000000</double>
      </property>
     </widget>
    </item>
    <item row="19" column="4">
     <widget class="QProgressBar" name="Progress">
      <property name="maximumSize">
       <size>
        <width>16777215</width>
        <height>16777215</height>
       </size>
      </property>
      <property name="value">
       <number>0</number>
      </property>
     </widget>
    </item>
    <item row="0" column="3" rowspan="20">
     <widget class="Line" name="line">
      <property name="orientation">
       <enum>Qt::Vertical</enum>
      </property>
     </widget>
    </item>
    <item row="9" column="0" rowspan="11" colspan="3">
     <spacer name="verticalSpacer">
      <property name="orientation">
       <enum>Qt::Vertical</enum>
      </property>
      <property name="sizeHint" stdset="0">
       <size>
        <width>20</width>
        <height>40</height>
       </size>
      </property>
     </spacer>
    </item>
    <item row="0" column="4" rowspan="19">
     <widget class="QPlainTextEdit" name="preview">
      <property name="minimumSize">
       <size>
        <width>400</width>
        <height>0</height>
       </size>
      </property>
      <property name="readOnly">
       <bool>true</bool>
      </property>
     </widget>
    </item>
    <item row="1" column="0">
     <widget class="QLineEdit" name="GeometryPath_ln"/>
    </item>
    <item row="6" column="0" colspan="3">
     <widget class="QPushButton" name="Preview_btn">
      <property name="text">
       <string>Process</string>
      </property>
     </widget>
    </item>
    <item row="5" column="0" colspan="3">
     <widget class="QDoubleSpinBox" name="TotalHeight_num">
      <property name="minimum">
       <double>1.000000000000000</double>
      </property>
      <property name="maximum">
       <double>900.000000000000000</double>
      </property>
     </widget>
    </item>
    <item row="4" column="0">
     <widget class="QLabel" name="lbl_TotalHeight">
      <property name="text">
       <string>Total extrusion height (mm)</string>
      </property>
     </widget>
    </item>
    <item row="2" column="0">
     <widget class="QLabel" name="label_4">
      <property name="text">
       <string>Layer height (mm)</string>
      </property>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menuBar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>803</width>
     <height>18</height>
    </rect>
   </property>
   <widget class="QMenu" name="menuFile">
    <property name="title">
     <string>File</string>
    </property>
    <addaction name="actionQuit"/>
   </widget>
   <addaction name="menuFile"/>
  </widget>
  <widget class="QStatusBar" name="statusBar"/>
  <action name="actionQuit">
   <property name="text">
    <string>Quit</string>
   </property>
  </action>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <resources/>
 <connections>
  <connection>
   <sender>SelectFileGeom_btn</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>OpenGeometryFile()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>137</x>
     <y>55</y>
    </hint>
    <hint type="destinationlabel">
     <x>226</x>
     <y>159</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>Preview_btn</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>Process()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>46</x>
     <y>128</y>
    </hint>
    <hint type="destinationlabel">
     <x>491</x>
     <y>183</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>Layer_num</sender>
   <signal>valueChanged(double)</signal>
   <receiver>MainWindow</receiver>
   <slot>setTotalHeightMin(double)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>104</x>
     <y>100</y>
    </hint>
    <hint type="destinationlabel">
     <x>491</x>
     <y>183</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>Layer_num</sender>
   <signal>valueChanged(double)</signal>
   <receiver>MainWindow</receiver>
   <slot>UpdateStats()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>104</x>
     <y>100</y>
    </hint>
    <hint type="destinationlabel">
     <x>491</x>
     <y>183</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>TotalHeight_num</sender>
   <signal>valueChanged(double)</signal>
   <receiver>MainWindow</receiver>
   <slot>UpdateStats()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>104</x>
     <y>145</y>
    </hint>
    <hint type="destinationlabel">
     <x>491</x>
     <y>183</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionQuit</sender>
   <signal>triggered()</signal>
   <receiver>MainWindow</receiver>
   <slot>close()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>401</x>
     <y>257</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <slot>OpenGeometryFile()</slot>
  <slot>Process()</slot>
  <slot>setTotalHeightMin(double)</slot>
  <slot>UpdateStats()</slot>
 </slots>
</ui>
